### PR TITLE
compiler: support nested struct selectors

### LINKED ISSUE #957: Add support of nested selectors
One of the common ways to interact with slice of structures is to iterate over them and access structure fields. 

```go
for i := 0; i < len(nodes); i++ {
    foo(nodes[i].id)
}
```

This code doesn't compile with an error: `nested selectors not supported yet`. You have to change the code.

```go
for i := 0; i < len(nodes); i++ {
    node := nodes[i]
    foo(node.id)
}
```

I think support of nested selectors will be quite useful for smart-contract developers.

### PR DESCRIPTION
**Pull request by fyrchik**
Closes #957 .

### CODE REVIEW

File: pkg/compiler/struct_test.go [hunk 1]
```diff
@@ -338,8 +338,23 @@ var structTestCases = []testCase{
...
+			return s1.y.a
+		}`,
+		big.NewInt(3),
+	},
 }
 
 func TestStructs(t *testing.T) {
-	runTestCases(t, structTestCases)
+	l := len(structTestCases)
+	runTestCases(t, structTestCases[l-1:])
```
Comment (roman-khimov) [inline pkg/compiler/struct_test.go]: ?

File: pkg/compiler/struct_test.go [hunk 2]
```diff
@@ -352,9 +352,24 @@ var structTestCases = []testCase{
...
+			return s1.x.a
+		}`,
+		big.NewInt(11),
+	},
 }
 
 func TestStructs(t *testing.T) {
-	l := len(structTestCases)
-	runTestCases(t, structTestCases[l-1:])
+	runTestCases(t, structTestCases)
```
Comment (roman-khimov) [inline pkg/compiler/struct_test.go]: :eyes:

File: pkg/compiler/struct_test.go (hunk 1 above)
Comment (fyrchik) [reply pkg/compiler/struct_test.go]: oh

File: pkg/compiler/struct_test.go (hunk 2 above)
Comment (fyrchik) [reply pkg/compiler/struct_test.go]: Fixed.

**Pull request merged**